tests/unittest_nodes.py

```diff
@@ -1371,5 +1371,181 @@ def test():
     assert bool(node.is_generator())
 
 
+@pytest.mark.skipif(
+    sys.version_info < (3, 10), reason="pattern matching was added in PY310"
+)
+class TestMatch:
+    @staticmethod
+    def test_match_simple():
+        node = builder.extract_node(
+            """
+        match status:
+            case 200:
+                pass
+            case 401 | 402 | 403:
+                pass
+            case None:
+                pass
+            case _:
+                pass
+        """
+        )
+        assert isinstance(node, nodes.Match)
+        assert isinstance(node.subject, nodes.Name)
+        assert isinstance(node.cases, list) and len(node.cases) == 4
+        assert node.subject.name == "status"
+        case0, case1, case2, case3 = node.cases
+
+        assert isinstance(case0.pattern, nodes.MatchValue)
+        assert case0.guard is None
+        assert isinstance(case0.body[0], astroid.Pass)
+        assert (
+            isinstance(case0.pattern.value, astroid.Const)
+            and case0.pattern.value.value == 200
+        )
```

Comment:
Idem here. I think we can group the lines dealing with `pattern`. IMHO it would make more sense but maybe i'm nitpicking.